zcash_client_backend/src/data_api/chain.rs

```diff
@@ -230,26 +334,27 @@ where
                     up.mark_spent(tx_row, &spend.nf)?;
                 }
 
-                nullifiers.retain(|(nf, _acc)| {
-                    tx.shielded_spends
-                        .iter()
-                        .find(|spend| &spend.nf == nf)
-                        .is_none()
-                });
+                // remove spent nullifiers from the nullifier set
+                nullifiers
+                    .retain(|(nf, _acc)| !tx.shielded_spends.iter().any(|spend| &spend.nf == nf));
 
                 for output in tx.shielded_outputs {
-                    let nf = output.note.nf(
-                        &extfvks[output.account].fvk.vk,
-                        output.witness.position() as u64,
-                    );
+                    match &extfvks.get(&output.account) {
+                        Some(extfvk) => {
```

Comment:
Since you're doing nothing in the `None` case:
```suggestion
                    if let Some(extfvk) = &extfvks.get(&output.account) {
```